exercises/gagik_vardanyan/ex02_oo_basics/src/include/Square.h

```diff
@@ -4,18 +4,17 @@
 
 #ifndef CLASS_SQUARE
 #define CLASS_SQUARE
-#include "ShapeBase.h"
+#include "Shape.h"
 
-class Square: public ShapeBase {
+class Square: public Shape {
 private:
     double side; /**< side of the square */
 
-protected:
+public:
     ///clone function
     /// @return new square object
     Square* clone() const {return new Square(*this);}
```

Comment:
Good. In c++ you can also take advantage of "Covariant return types", so you could have declared the `clone` to be:

```cpp
 Square* clone() const {return new Square(*this);}
```

That would have been slightly better, but it's a minor tweak to make.
The clone implementation should be in the cpp, but generally you've done a good job moving things into cpp files.